bundler/helpers/v2/lib/functions/force_updater.rb

```diff
@@ -53,6 +62,28 @@ def run
                 :update_multiple_dependencies
     alias update_multiple_dependencies? update_multiple_dependencies
 
+    def add_extra_top_level_deps(specs, updates)
+      top_level_deps.each do |dep|
+        name = dep.name
+
+        original_version = original_specs.find { |s| s.name == name }&.version
+        new_version = specs[name].first&.version
+        next unless original_version != new_version
+
+        dependency_of_updated_dep = updates.any? do |updated_dep|
+          new_spec = specs[updated_dep].first
+          next false unless new_spec
```

Comment:
How does a dependency end up in the updates list if it's not in the spec? Would this be a peer dependency, for example?

Reply:
Nice catch, I don't think that can happen. One case where a transitve dependency would no longer be in the resolved array of specs is when it's removed, but since here we know it's a dependency that got explictly updated, I don't think it can happen. Let me simplify this.

Reply:
I pushed a simpler alternative that I believe should also be more correct. Technically we should be checking all matching specs, not just the first (for the same name and version, there could be different gems resolved for the different platforms, and they could even have different dependencies).